src/signing_key.rs

```diff
@@ -106,6 +111,59 @@ impl From<[u8; 32]> for SigningKey {
     }
 }
 
+impl<'a> TryFrom<PrivateKeyInfo<'a>> for SigningKey {
+    type Error = Error;
+    fn try_from(pki: PrivateKeyInfo) -> Result<Self, Error> {
+        assert_eq!(pki.algorithm.oid, "1.3.101.112".parse().unwrap());
+        assert_eq!(pki.algorithm.parameters, None);
+
+        SigningKey::try_from(pki.private_key)
```

Comment:
Note that the `assert_eq!` calls as well as `unwrap` will panic on error. You'll probably want to return an `Error` in this case:

```suggestion
        match pki.algorithm {
            AlgorithmIdentifier { oid: OID, parameters: None } => SigningKey::try_from(pki.private_key),
            _ => Err(Error)
        }
```

Reply:
Ahhh, sorry, I got in a hurry and misunderstood what you wrote. That suggestion looks great.

Reply:
Thanks for all the help you've been providing! Still learning my way around Rust. This entire process really helps.